src/circuit/circuit_frame.rs

```diff
@@ -3055,6 +3151,168 @@ fn apply_continuation<F: LurkField, CS: ConstraintSystem<F>>(
 
         let res = AllocatedPtr::from_parts(&res_tag, &val);
 
+        // Next constraints are used for number comparisons: <, <=, >, >=
+        ///////////////////////////////////////////////////////////////////////
+        let double_a = constraints::add(&mut cs.namespace(|| "double a"), a, a)?;
+        // Ideally we would compute the bit decomposition for a, not for 2a,
+        // since it would be possible to use it for future purposes.
+        // TODO: replace bit decompositon from 2a to a.
+        let double_a_bits = double_a
+            .to_bits_le_strict(&mut cs.namespace(|| "double a lsb"))
+            .unwrap();
+        let lsb_2a = double_a_bits.get(0);
+
+        let double_b = constraints::add(&mut cs.namespace(|| "double b"), b, b)?;
+        // TODO: replace bit decompositon from 2b to b.
+        let double_b_bits = double_b
+            .to_bits_le_strict(&mut cs.namespace(|| "double b lsb"))
+            .unwrap();
+        let lsb_2b = double_b_bits.get(0);
+
+        let diff_is_zero = alloc_is_zero(&mut cs.namespace(|| "diff is zero"), &diff)?;
+        let double_diff = constraints::add(&mut cs.namespace(|| "double diff"), &diff, &diff)?;
+        // TODO: replace bit decompositon from 2diff to diff.
+        let double_diff_bits = double_diff.to_bits_le_strict(&mut cs).unwrap();
+        let lsb_2diff = double_diff_bits.get(0);
+
+        // We have that a number is defined to be negative if the parity bit (the
+        // least significant bit) is odd after doubling, meaning that the field element
+        // (after doubling) is larger than the underlying prime p that defines the
+        // field, then a modular reduction must be carried out, changing the parity that
+        // should be even (since we multiplied by 2) to odd. In other words, we define
+        // negative numbers to be those field elements that are larger than p/2.
+        let a_is_negative = lsb_2a.unwrap();
+        let b_is_negative = lsb_2b.unwrap();
+        let diff_is_negative = lsb_2diff.unwrap();
+
+        let diff_is_negative_or_zero = constraints::or(
+            &mut cs.namespace(|| "diff is negative or zero"),
+            lsb_2diff.unwrap(),
+            &diff_is_zero,
+        )?;
+
+        let diff_is_strictly_positive = Boolean::and(
+            &mut cs.namespace(|| "diff is strictly positive"),
+            &diff_is_negative.not(),
+            &Boolean::not(&diff_is_zero),
+        )?;
+
+        let diff_is_positive_or_zero = constraints::or(
+            &mut cs.namespace(|| "diff is positive or zero"),
+            &diff_is_negative.not(),
+            &diff_is_zero,
+        )?;
```

Comment:
Isn't 'positive or zero' *exactly* equivalent to 'not negative'? If so (as I believe), then this `or` is unnecessary.

Put differently: if `diff` is zero, then `2 * diff` is also zero and is therefore even, so `diff_is_negative.not()` will always be true. Therefore, we don't need to check the `diff_is_zero` case specially. It's already handled correctly by virtue of zero not being negative.

Reply:
For simplicity and standardization (see other comments), I would call this `diff_is_not_negative` and implement as just `diff_is_negative.not()`.